### ISSUE TITLE
Issue with SGE cluster that automatically loads environment variables

### ISSUE THREAD

**Issue by amiraa127** [Illumina/pyflow#14]
Hi,

I've been getting errors running pyflow on a SGE cluster that automatically loads environment variables:

```
[2015-11-13T18:29:59.385376] [node105.cm.cluster] [90604_1] [TaskManager] [ERROR] [Isis_EZTn5-10nM-20ng] [taskWrapper-stderr] sh: module: line 1: syntax error: unexpected end of file
[2015-11-13T18:29:59.387136] [node105.cm.cluster] [90604_1] [TaskManager] [ERROR] [Isis_EZTn5-10nM-20ng] [taskWrapper-stderr] sh: error importing function definition for `BASH_FUNC_module'
[2015-11-13T18:29:59.389185] [node105.cm.cluster] [90604_1] [TaskManager] [ERROR] [Isis_EZTn5-10nM-20ng] [taskWrapper-stderr] sh: ml: line 1: syntax error: unexpected end of file
[2015-11-13T18:29:59.392445] [node105.cm.cluster] [90604_1] [TaskManager] [ERROR] [Isis_EZTn5-10nM-20ng] [taskWrapper-stderr] sh: error importing function definition for `BASH_FUNC_ml'
[2015-11-13T18:29:59.394684] [node105.cm.cluster] [90604_1] [TaskManager] [ERROR] [Isis_EZTn5-10nM-20ng] [taskWrapper-stderr] [2015-11-13T18:26:54.046586] [node031.cm.cluster] [90604_1] [py
flowTaskWrapper:Isis_EZTn5-10nM-20ng] [wrapperSignal] taskStart
[2015-11-13T18:29:59.398157] [node105.cm.cluster] [90604_1] [TaskManager] [ERROR] [Isis_EZTn5-10nM-20ng] [taskWrapper-stderr] [2015-11-13T18:29:21.829184] [node031.cm.cluster] [90604_1] [py
flowTaskWrapper:Isis_EZTn5-10nM-20ng] [wrapperSignal] taskExitCode 0
[2015-11-13T18:29:59.400575] [node105.cm.cluster] [90604_1] [TaskManager] [ERROR] [Isis_EZTn5-10nM-20ng] [taskWrapper-stderr] [2015-11-13T18:29:21.834988] [node031.cm.cluster] [90604_1] [py
flowTaskWrapper:Isis_EZTn5-10nM-20ng] [wrapperSignal] taskElapsedSec 147
[2015-11-13T18:29:59.402592] [node105.cm.cluster] [90604_1] [TaskManager] [ERROR] Shutting down task submission. Waiting for remaining tasks to complete.
```

I've noticed that removing the '-V' option from qsub resolves this issue. Is there any way (using the client api) that qsub be invoked without the '-V' option?

**Comment by hyjkim** [comment]
Hey,

I was also having this problem. The error only occurs when combining the '-V' option in qsub with the python 'platform.system()' command. I believe this issue is related to a patch addressing the shellshock exploit in bash.

The workaround I am using is to set isWin=False rather than isWin=isWindows() on line 264 of pyflowTaskWrapper.py. You will still see similar errors in your stderr.log file, but they will not cause your workflows to crash.

Best,
John

**Comment by ctsa** [comment]
Closed @ commit: 8da11246b2a8c88b32932662d7b44cf0a2823674
release: https://github.com/Illumina/pyflow/releases/tag/v1.1.12
